packages/flutter/lib/src/scheduler/ticker.dart

```diff
@@ -12,80 +15,233 @@ import 'binding.dart';
+  // and then this could return an accurate view of the actual scheduler.
+  bool get isTicking => _completer != null && !muted;
+
+  /// Whether time is elapsing for this ticker. Becomes true when [start] is
+  /// called and false when [stop] is called.
+  ///
+  /// A ticker can be active yet not be actually ticking (i.e. not be calling
+  /// the callback). To determine if a ticker is actually ticking, use
+  /// [isTicking].
+  bool get isActive => _completer != null;
 
-  /// Starts calling the ticker's callback once per animation frame.
+  Duration _startTime;
+
+  /// Starts the clock for this ticker. If the ticker is not [muted], then this
+  /// also starts calling the ticker's callback once per animation frame.
+  ///
+  /// The returned future resolves once the ticker [stop]s ticking.
+  ///
+  /// Calling this sets [isActive] to true.
   ///
-  /// The returned future resolves once the ticker stops ticking.
+  /// This method cannot be called while the ticker is active. To restart the
+  /// ticker, first [stop] it.
   Future<Null> start() {
     assert(!isTicking);
     assert(_startTime == null);
     _completer = new Completer<Null>();
-    _scheduleTick();
-    if (SchedulerBinding.instance.isProducingFrame)
+    if (shouldScheduleTick)
+      scheduleTick();
+    if (SchedulerBinding.instance.schedulerPhase.index > SchedulerPhase.idle.index &&
+        SchedulerBinding.instance.schedulerPhase.index < SchedulerPhase.postFrameCallbacks.index)
       _startTime = SchedulerBinding.instance.currentFrameTimeStamp;
     return _completer.future;
   }
 
   /// Stops calling the ticker's callback.
   ///
   /// Causes the future returned by [start] to resolve.
+  ///
+  /// Calling this sets [isActive] to false.
+  ///
+  /// This method does nothing if called when the ticker is inactive.
   void stop() {
     if (!isTicking)
       return;
 
-    _startTime = null;
-
-    if (_animationId != null) {
-      SchedulerBinding.instance.cancelFrameCallbackWithId(_animationId);
-      _animationId = null;
-    }
-
     // We take the _completer into a local variable so that isTicking is false
-    // when we actually complete the future (isTicking uses _completer
-    // to determine its state).
+    // when we actually complete the future (isTicking uses _completer to
+    // determine its state).
     Completer<Null> localCompleter = _completer;
     _completer = null;
+    _startTime = null;
     assert(!isTicking);
+
+    unscheduleTick();
     localCompleter.complete();
   }
 
+
+  final TickerCallback _onTick;
+
+  int _animationId;
+
+  /// Whether this ticker has already scheduled a frame callback.
+  @protected
+  bool get scheduled => _animationId != null;
+
+  /// Whether a tick should be scheduled.
+  ///
+  /// If this is true, then calling [scheduleTick] should succeed.
+  ///
+  /// Reasons why a tick should not be scheduled include:
+  ///
+  /// * A tick has already been scheduled for the coming frame.
+  ///
```

Comment:
I would remove the empty lines here since it makes reading the comment in plain text harder.